kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopEventManager.java

```diff
@@ -0,0 +1,291 @@
+        }
+
+    }
+
+
+    public void put(KopEvent event) {
+        try {
+            putLock.lock();
+            queue.put(event);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            log.error("Error put event {} to coordinator event queue {}", event, e);
+        } finally {
+            putLock.unlock();
+        }
+    }
+
+    public void clearAndPut(KopEvent event) {
+        try {
+            putLock.lock();
+            queue.clear();
+            put(event);
+        } finally {
+            putLock.unlock();
+        }
+    }
+
+    static class KopEventThread extends ShutdownableThread {
+
+        public KopEventThread(String name) {
+            super(name);
+        }
+
+        @Override
+        protected void doWork() {
+            KopEvent event = null;
+            try {
+                event = queue.take();
+                event.process();
+            } catch (InterruptedException e) {
+                log.error("Error processing event {}, {}", event, e);
+            }
+        }
+
+    }
+
+    private void registerChildChangeHandler() {
+        metadataStore.registerListener(this::handleChildChangePathNotification);
+
+        // Really register ChildChange notification.
+        metadataStore.getChildren(getDeleteTopicsPath());
+        // init local kop brokers cache
+        getBrokers(metadataStore.getChildren(getBrokersChangePath()).join());
+    }
+
+    private void handleChildChangePathNotification(Notification notification) {
+        if (notification.getPath().equals(LoadManager.LOADBALANCE_BROKERS_ROOT)) {
+            this.brokersChangeHandler.handleChildChange();
+        } else if (notification.getPath().equals(getDeleteTopicsPath())) {
+            this.deletionTopicsHandler.handleChildChange();
+        }
+    }
+
+    private void getBrokers(List<String> pulsarBrokers) {
+        HashSet<Node> kopBrokers = Sets.newHashSet();
+        pulsarBrokers.forEach(broker -> {
+            try {
+                Optional<GetResult> brokerData = metadataStore.get(
+                        getBrokersChangePath() + "/" + broker).join();
+
+                if (brokerData.isPresent()) {
+                    JsonObject jsonObject = parseJsonObject(
+                            new String(brokerData.get().getValue(), StandardCharsets.UTF_8));
+                    JsonObject protocols = jsonObject.getAsJsonObject("protocols");
+                    JsonElement element = protocols.get("kafka");
+
+                    if (element != null) {
+                        String kopBrokerStr = element.getAsString();
+                        Node kopNode = getNode(kopBrokerStr);
+                        kopBrokers.add(kopNode);
+                    }
+                }
+            } catch (Exception e) {
+                log.error("Get broker {} path data failed which have an error {}", broker, e.getMessage());
+                e.printStackTrace();
```

Comment:
If you want to print the stack trace, use

```java
log.error("Get broker {} path data failed which have an error", broker, e);
```

Reply:
Thanks, I will do